x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/FieldExtractExec.java

```diff
@@ -57,14 +63,20 @@ public class FieldExtractExec extends UnaryExec implements EstimatesRowSize {
 
     private List<Attribute> lazyOutput;
 
-    public FieldExtractExec(Source source, PhysicalPlan child, List<Attribute> attributesToExtract) {
-        this(source, child, attributesToExtract, Set.of(), Set.of());
+    public FieldExtractExec(
+        Source source,
+        PhysicalPlan child,
+        List<Attribute> attributesToExtract,
+        MappedFieldType.FieldExtractPreference defaultPreference
```

Comment:
This constructor should not have the field-extract preference, since that is local-physical planning only. It should set the default NONE on the call to this() below.

Reply:
Sure.

Reply:
Just tried to apply this and we do need it, at least we need it how things are shaped now. `InsertFieldExtraction` wants to make a `FieldExtractExec` on the data nodes with this already set.